--- a/src/wayland/shm/handlers.rs
+++ b/src/wayland/shm/handlers.rs
@@ -185,6 +185,7 @@
             Request::Resize { size } => {
                 if size <= 0 {
                     pool.post_error(wl_shm::Error::InvalidFd, "invalid wl_shm_pool size");
+                    return;
                 }
 
                 if let Err(err) = arc_pool.resize(NonZeroUsize::try_from(size as usize).unwrap()) {
